pkg/compactor/compactor_test.go

```diff
@@ -1019,7 +1019,7 @@ func removeIgnoredLogs(input []string) []string {
 
 	for i := 0; i < len(input); i++ {
 		log := input[i]
-		if strings.Contains(log, "block.MetaFetcher") || strings.Contains(log, "block.BaseFetcher") {
+		if strings.Contains(log, "block.MetaFetcher") || strings.Contains(log, "block.BaseFetcher") || strings.Contains(log, "failed deleting non-compaction group directories/files, some disk space usage might have leaked") {
```

Comment:
This doesn't sound correct. We shouldn't filter out error logs. If there's an issue we should fix it. Could you investigate, please?

_Suggestion: try to upgrade to Thanos `master` again. The compactor local filesystem cleanup is something still under work._

Reply:
Hi @pracucci.

I did upgrade Thanos and the problem persists. This log is `logged` as an `Error` but as an `Warn`.

Investigating this issue seems that the problem is cause during the first compaction the `compactDir` is not yet created and then we get an error - no such file or directory when trying to delete on [this](https://github.com/thanos-io/thanos/blob/1bbad3b5053d28d6b0f20c83998ed9702d5e042e/pkg/compact/compact.go#L964) line.

We can see that the folder is immediately created when the first compaction run from [here](https://github.com/thanos-io/thanos/blob/1bbad3b5053d28d6b0f20c83998ed9702d5e042e/pkg/compact/compact.go#L977), [here](https://github.com/thanos-io/thanos/blob/1bbad3b5053d28d6b0f20c83998ed9702d5e042e/pkg/compact/compact.go#L917) and [here](https://github.com/thanos-io/thanos/blob/1bbad3b5053d28d6b0f20c83998ed9702d5e042e/pkg/compact/compact.go#L503)

I think its fine to ignore this new log line there or we should try to do a change on thanos to not log that warn in case of a `ENOENT`. Thoughts?

Reply:
This looks harmless to me. Changing thanos code to not generate error if dir doesn't exist would be another way of fixing this.

Reply:
I disagree on not fixing it (and ignoring the log) because of a couple of reasons:
1. If a Cortex user sees this warning, it may be "warned" even if it's a non issue
2. If in the future other issues cause this deletion to fail, we'll not notice it if we ignore such log

I agree it's "not a big deal" right now, but given the fix looks trivial we could just go ahead and fix it:
https://github.com/thanos-io/thanos/pull/3974

Reply:
Thanks!!! And thanks for following up with thanos! :)

Reply:
https://github.com/thanos-io/thanos/pull/3974 has been merged so if you upgrade Thanos again (which would be required anyway because of fixes done in the meanwhile) you can then remove the change from `removeIgnoredLogs()`. Thanks!